docs/source/daml/reference/contract-keys.rst

```diff
@@ -62,6 +62,17 @@ Like ``fetch``, ``fetchByKey`` needs to be authorized by at least one stakeholde
 
 This means that if it fails, it doesn't guarantee that a contract with that key doesn't exist, just that the submitting Party doesn't know about it, or there are issues with authorization.
 
+.. _doescontractexistforkey:
+
+canSeeContractWithKey
+=====================
+
+``boolean <- canSeeContractWithKey contractKey``
+
+Use ``canSeeContractWithKey`` to check whether, in the current context, you can see an active contract for the current key. If the contract exists and the current context has the permission to see it, returns `True`, otherwise returns `False`.
+
+``canSeeContractWithKey`` needs to be authorized by **all** maintainers of the contract you are trying to lookup. The reason for that is denial of service protection. Without this restriction, a malicious Party could legitimately request another Party to validate any number of negative contract key lookups without that other Party ever having signed anything.
```

Comment:
```suggestion
``canSeeContractWithKey`` needs to be authorized by **all** maintainers of the contract you are trying to lookup. The reason for that is denial of service protection. Without this restriction, a malicious party could legitimately request another Party to validate any number of negative contract key lookups without that other party ever having signed anything.
```

Reply:
Text substantially changed and does not contain "party" anymore. Note that "Party" is capitalized in both `lookupByKey` and `fetchByKey` sections (untouched by this PR); should I update those?

Reply:
@bame-da Do you remember if there was a specific reason to capitalize the word "party" in some sections? Or was it an oversight?

Reply:
I think it's to distinguish party in the loose sense against Party in the technical sense. Of course, a Party can't be malicious as it's a technical context, not do Parties validate anything so I don't see a good case for capitalizing here.

Reply:
All existing cases of "Party" are in the context of "the submitting Party", so I guess given the explanation it makes sense to keep the capitalization.